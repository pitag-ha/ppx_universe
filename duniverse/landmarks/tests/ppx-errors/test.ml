[@@@landmark "random string"];;

module M = struct
  [@@@landmark "random string"];;
end;;

[@@@landmark "auto" "no auto"];;

let () = (print_endline "hello")[@landmark "too"][@landmark "many"][@landmark "landmarks"];;

let[@landmark] () = print_endline "hello";;

let () =
  let[@landmark] _ = 2+2 in ();;

