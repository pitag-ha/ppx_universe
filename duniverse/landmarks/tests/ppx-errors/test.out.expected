Line 1, characters 0-29:
1 | [@@@landmark "random string"];;
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: ppx_landmark: expecting payload in ["auto","auto-off","disable"]
Line 3, characters 2-31:
3 |   [@@@landmark "random string"];;
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: ppx_landmark: expecting payload in ["auto","auto-off"]
Line 2, characters 0-30:
2 | [@@@landmark "auto" "no auto"];;
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: ppx_landmark: payload is not a string
Line 2, characters 9-32:
2 | let () = (print_endline "hello")[@landmark "too"][@landmark "many"][@landmark "landmarks"];;
             ^^^^^^^^^^^^^^^^^^^^^^^
Error: ppx_landmark: too many attributes
Line 2, characters 0-41:
2 | let[@landmark] () = print_endline "hello";;
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: ppx_landmark: this landmark annotation requires a name argument
Line 3, characters 2-24:
3 |   let[@landmark] _ = 2+2 in ();;
      ^^^^^^^^^^^^^^^^^^^^^^
Error: ppx_landmark: this landmark annotation requires a name argument

